### ISSUE TITLE
Kubernetes only works with StatefulSet ( not deployments or daemonsets )

### ISSUE THREAD

**Issue by qrpike** [YugaByte/yugabyte-db#1608]
So YugaByte works fine when using stateful sets, this is because each pod gets it's own service which matches the pods name. Eg: Pod name `yb-master-0` is the pod name, and is accessible to other pods from that DNS record. 

When using a Deployment, or DaemonSet the pod names are randomized. I creating a service for each pod, for example `master-0` which points to pod `master-0-xvldsn`. I set the cli param `--server_broadcast_addresses=master-0.namespace:7100` to make sure the other nodes would connect via the service name `master-0` instead of trying to connect directly to the pod. However the other nodes continue to try to connect using the actual pod name `master-0-xvldsn`.

This also happens for the server nodes as well. The CQL command fails because it is also try to connect using the pod name. 

This is important for our environment where we need to manually schedule which nodes will be place where, and assign rack / region params.

**Comment by rkarthik007** [comment]
Hi @qrpike,

Great question, thanks for posting. I think our docs could benefit from this as well.

The way you describe should theoretically work as long as you specify the service names for the all the `--master_addresses` parameters, and set the `broadcast_address`. Could you please post your yaml? Mainly looking for the `cmd` for the master and tserver pods.

@Arnav15 - could you please give this a try?

**Comment by qrpike** [comment]
So I was trying to replicate the setup this morning, I deleted all deployments, services, etc, I wiped all machines of the mapped local directory, and even scheduled them on completely different machines. And I keep getting the following error like there is some kind of leftover state:

```
I0624 15:43:16.872499     1 master_main.cc:93] NumCPUs determined to be: 32
I0624 15:43:16.879556     1 server_base_options.cc:209] Resolved master addresses: [10.3.81.166:7100, 10.3.216.221:7100]
I0624 15:43:16.880220     1 mem_tracker.cc:242] MemTracker: hard memory limit is 12.592720 GB
I0624 15:43:16.880283     1 mem_tracker.cc:244] MemTracker: soft memory limit is 10.703812 GB
I0624 15:43:16.883890     1 master_main.cc:114] Initializing master server...
I0624 15:43:16.886893     1 fs_manager.cc:232] Opened local filesystem: /yugabyte
uuid: "81b31888aa6046d0b3bb5662337009ac"
format_stamp: "Formatted at 2019-06-24 15:37:12 on master2-59f5584d5c-g864v"
I0624 15:43:16.887671     1 server_base.cc:200] Auto setting FLAGS_num_reactor_threads to 16
I0624 15:43:16.895928     1 master_main.cc:117] Starting Master server...
I0624 15:43:16.900779     1 webserver.cc:147] Starting webserver on 0.0.0.0:7000
I0624 15:43:16.900833     1 webserver.cc:152] Document root: /home/yugabyte/www
I0624 15:43:16.901904     1 webserver.cc:239] Webserver started. Bound to: http://0.0.0.0:7000/
I0624 15:43:16.902609     1 rpc_server.cc:167] RPC server started. Bound to: 0.0.0.0:7100
I0624 15:43:16.917347    35 sys_catalog.cc:194] Trying to load previous SysCatalogTable data from disk
E0624 15:43:16.918123    35 master.cc:217] Master@0.0.0.0:7100: Unable to init master catalog manager: Not found (yb/util/env_posix.cc:1497): Unable to initialize catalog manager: Failed to initialize sys tables async: Could not load Raft group metadata from /yugabyte/yb-data/master/tablet-meta/00000000000000000000000000000000: /yugabyte/yb-data/master/tablet-meta/00000000000000000000000000000000: No such file or directory (error 2)
E0624 15:43:16.918279     1 main_util.cc:22] Not found (yb/util/env_posix.cc:1497): Unable to initialize catalog manager: Failed to initialize sys tables async: Could not load Raft group metadata from /yugabyte/yb-data/master/tablet-meta/00000000000000000000000000000000: /yugabyte/yb-data/master/tablet-meta/00000000000000000000000000000000: No such file or directory (error 2)
I0624 15:43:16.918401     1 master.cc:260] Master@0.0.0.0:7100 shutting down...
I0624 15:43:16.932934     1 master.cc:268] Master@0.0.0.0:7100 shutdown complete.
```

When accessing the nodes they are scheduled on, the /yugabyte/ directory has contents. I also chmod 777 the directory to make sure it wasn't a permissions issue.

**Comment by qrpike** [comment]
As for the actual YAML configurations, here are the templates:

### Masters:
```yaml

apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: master1
  namespace: yugabyte
  labels:
    app: yb-master
    server: master1
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: yb-master
        server: master1
    spec:
      restartPolicy: Always
      containers:
      - name: master1
        image: yugabytedb/yugabyte:1.2.10.0-b13
        command:
          - "/home/yugabyte/bin/yb-master"
          - "--fs_data_dirs=/yugabyte"
          - "--server_broadcast_addresses=master1.yugabyte:7100"
          - "--use_private_ip=never"
          - "--master_addresses=master1.yugabyte:7100,master2.yugabyte:7100,master3.yugabyte:7100"
          - "--master_replication_factor=3"
          - "--replication_factor=3"
          - "--logtostderr"
          - "--placement_zone=4152"
          - "--placement_region=NY2"
        ports:
        - containerPort: 7000
          name: ui
        - containerPort: 7100
          name: rpc-port
        resources:
          limits:
            cpu: 4
            memory: 6G
        volumeMounts:
        - name: datadir
          mountPath: /yugabyte
      volumes:
      - name: datadir
        hostPath:
          path: "/yugabyte-master/"
      nodeSelector:
        id: "1"

---

apiVersion: v1
kind: Service
metadata:
  name: master1
  namespace: yugabyte
  labels:
    app: yb-master
    server: master1
spec:
  ports:
  - name: ui
    port: 7000
  - name: rpc-port
    port: 7100
  selector:
    app: yb-master
    server: master1
```

### Servers:

```yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: server1
  namespace: yugabyte
  labels:
    app: yb-tserver
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: yb-tserver
        server: server1
    spec:
      restartPolicy: Always
      containers:
      - name: server1
        image: yugabytedb/yugabyte:1.2.10.0-b13
        command:
          - "/home/yugabyte/bin/yb-tserver"
          - "--fs_data_dirs=/yugabyte"
          - "--tserver_master_addrs=master1.yugabyte:7100,master2.yugabyte:7100,master3.yugabyte:7100"
          - "--tserver_master_replication_factor=3"
          - "--server_broadcast_addresses=server1.yugabyte:9100"
          - "--placement_zone=4152"
          - "--placement_region=NY2"
          - "--logtostderr"
        ports:
        - containerPort: 9000
          name: tserver-ui
        - containerPort: 9100
          name: tserver-rpc
        - containerPort: 9042
          name: cassandra
        - containerPort: 6379
          name: redis
        volumeMounts:
        - name: datadir
          mountPath: /yugabyte
        resources:
          limits:
            cpu: 6
            memory: 6G
      volumes:
      - name: datadir
        hostPath:
          path: "/yugabyte"
      nodeSelector:
        id: "10"

---

apiVersion: v1
kind: Service
metadata:
  name: server1
  namespace: yugabyte
  labels:
    app: yb-tserver
    server: server1
spec:
  ports:
  - name: ui
    port: 9000
  - name: rpc-port
    port: 9100
  - name: cassandra
    port: 9042
  - name: redis
    port: 6379
  selector:
    app: yb-tserver
    server: server1
```

**Comment by rkarthik007** [comment]
Hi @qrpike,

Thanks for posting your yaml.

**Could you please try the following:**

* Delete all contents of `/yugabyte` (specifically the `/yugabyte/yb-data` directory) from all the pods
* Once the above is done, restart the cluster

## Explanation

We have seen this happen if you "restart" YugaByte DB on top of existing data (since `/yugabyte` already has data) - when the first start was not successful. This is because the first unsuccessful start will leave partial information, causing the server to fail with errors such as 
```
Unable to initialize catalog manager: 
  Failed to initialize sys tables async:
    Could not load Raft group metadata from /yugabyte/yb-data/master/tablet-meta/00000000000000000000000000000000
```

With `StatefulSet`s, this is automatically taken care of because each create carves out a separate mount path. But in this case, it is mapped to the same location.

**Comment by qrpike** [comment]
For some reason I had to change the mount paths, even after removing all pods and all files `sudo rm -rf /yugabyte-master/`. Changing the mount path to `yb-master` now works. 

When viewing the UI which is a server which targets any master, I get mixed results, sometimes I get: 
![image](https://user-images.githubusercontent.com/669468/60035510-fd381b80-967a-11e9-8211-0c162d9cc91e.png)

And sometimes I get: 
![image](https://user-images.githubusercontent.com/669468/60035531-0aeda100-967b-11e9-8b30-4c744d2f61f6.png)

This also happens for the tablets tab: 
![image](https://user-images.githubusercontent.com/669468/60035597-38d2e580-967b-11e9-8539-138d1ee7e52d.png)

And sometimes I get: 
![image](https://user-images.githubusercontent.com/669468/60035609-438d7a80-967b-11e9-8bbc-6c45465f2a2e.png)

It seems to be referencing the pod name, instead of using the `server_broadcast_addresses` which is being set.

**Comment by qrpike** [comment]
This is the same issue I see when try to use the servers as well, when attempting to run cqlsh in the pod: 

```
[root@server1-74774bb7cd-x56rk yugabyte]# ./bin/cqlsh
Connection error: ('Unable to connect to any servers', {'127.0.0.1': <Error from server: code=0000 [Server error] message="Server Error. Resolve failed server1-74774bb7cd-x56rk: Host not found (non-authoritative), try again later
SELECT * FROM system.local WHERE key='local'
^^^^^^
 (error -2)">})
```

It's also trying to use the pod name, instead of the broadcasted address which is being set.

**Comment by Arnav15** [comment]
Hey @qrpike! I apologize for the delay in response. I got yugabyte working using the YAML you provided with some minor modifications. I added an RPC bind address so that the pods bind address resolves to the pod IP which can then be reached by the service. I also made a modification to the master address list since we need to check that the bind address of a master is present in the list of masters. I have filed an issue to work on [that](https://github.com/YugaByte/yugabyte-db/issues/1621), but for now you can get around that limitation by using the following YAMLs.

https://gist.github.com/Arnav15/3e337e815dcae095c95b3501edbaee7b
https://gist.github.com/Arnav15/9b7cc4ceeb7c4f39777e0c8f78735eca

Please let me know if you have any other questions or you run into any other issues!

**Comment by Arnav15** [comment]
@qrpike Just wanted to sync up again to see if the above YAMLs were helpful or if you needed anything else to get started with YB.

**Comment by qrpike** [comment]
@Arnav15 Yes, it seems to have solved the issue I was running into. I decided to go with a stateful set using network disks for the masters. Is this advisable, or should the masters also be using locally mounted SSDs?

**Comment by rkarthik007** [comment]
@qrpike - stateful sets would work well, don't think you would run into issues since the master is quite light on the io and processing. A lot of our other users are using it this way.

**Comment by qrpike** [comment]
It seems something still isn't quite right. I can create a keyspace, however if I try to create a table in a keyspace, I get a timeout, regardless of which node I'm on. 

When looking at the logs, it seems like it just keeps attempting a re-election. 

```
I0701 17:10:11.023422   175 leader_election.cc:240] T 11fcc1c5124d48ca99c758e4bd395ba6 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Election decided. Result: candidate lost.
I0701 17:10:11.023628  1067 raft_consensus.cc:2876] T 11fcc1c5124d48ca99c758e4bd395ba6 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 3.399s
I0701 17:10:11.023701  1067 raft_consensus.cc:2722] T 11fcc1c5124d48ca99c758e4bd395ba6 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Leader Pre-election lost for term 1. Reason: None given. Originator: 
I0701 17:10:11.200372  1068 raft_consensus.cc:813] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: ReportFailDetected: Starting NORMAL_ELECTION...
I0701 17:10:11.200438  1068 raft_consensus.cc:496] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Triggering leader pre-election, mode=NORMAL_ELECTION
I0701 17:10:11.200484  1068 raft_consensus.cc:2876] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 5.237s
I0701 17:10:11.200680  1068 raft_consensus.cc:535] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Starting pre-election with config: opid_index: -1 peers { permanent_uuid: "012245bfde494e12b856245c1b45022c" member_type: VOTER last_known_private_addr { host: "10.2.43.61" port: 9100 } last_known_broadcast_addr { host: "server1.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "35f6ba9096884be988424f4beb3353ea" member_type: VOTER last_known_private_addr { host: "10.2.80.72" port: 9100 } last_known_broadcast_addr { host: "server5.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "671457e0cf824f1bab6eefc5a8788d71" member_type: VOTER last_known_private_addr { host: "10.2.101.52" port: 9100 } last_known_broadcast_addr { host: "server3.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } }
I0701 17:10:11.200837  1068 leader_election.cc:215] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 35f6ba9096884be988424f4beb3353ea
I0701 17:10:11.200978  1068 leader_election.cc:215] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 671457e0cf824f1bab6eefc5a8788d71
W0701 17:10:11.201786   158 leader_election.cc:281] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 35f6ba9096884be988424f4beb3353ea: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: 6b69a7a766604b1497dc13efab6a0a21
W0701 17:10:11.201951   401 leader_election.cc:281] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 671457e0cf824f1bab6eefc5a8788d71: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: 6b69a7a766604b1497dc13efab6a0a21
I0701 17:10:11.201987   401 leader_election.cc:240] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Election decided. Result: candidate lost.
I0701 17:10:11.202117  1067 raft_consensus.cc:2876] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 3.379s
I0701 17:10:11.202181  1067 raft_consensus.cc:2722] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Leader Pre-election lost for term 1. Reason: None given. Originator: 
I0701 17:10:11.518371  1068 raft_consensus.cc:813] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: ReportFailDetected: Starting NORMAL_ELECTION...
I0701 17:10:11.518414  1068 raft_consensus.cc:496] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Triggering leader pre-election, mode=NORMAL_ELECTION
I0701 17:10:11.518446  1068 raft_consensus.cc:2876] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 5.120s
I0701 17:10:11.518546  1068 raft_consensus.cc:535] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Starting pre-election with config: opid_index: -1 peers { permanent_uuid: "66ef2b92cb5244af8c5398f7f3838fdb" member_type: VOTER last_known_private_addr { host: "10.2.55.104" port: 9100 } last_known_broadcast_addr { host: "server4.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "012245bfde494e12b856245c1b45022c" member_type: VOTER last_known_private_addr { host: "10.2.43.61" port: 9100 } last_known_broadcast_addr { host: "server1.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "671457e0cf824f1bab6eefc5a8788d71" member_type: VOTER last_known_private_addr { host: "10.2.101.52" port: 9100 } last_known_broadcast_addr { host: "server3.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } }
I0701 17:10:11.518657  1068 leader_election.cc:215] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 66ef2b92cb5244af8c5398f7f3838fdb
I0701 17:10:11.518729  1068 leader_election.cc:215] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 671457e0cf824f1bab6eefc5a8788d71
W0701 17:10:11.519656   172 leader_election.cc:281] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 66ef2b92cb5244af8c5398f7f3838fdb: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: c59a4e833445447aa25de7a34a5c03c7
W0701 17:10:11.519752   330 leader_election.cc:281] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 671457e0cf824f1bab6eefc5a8788d71: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: c59a4e833445447aa25de7a34a5c03c7
I0701 17:10:11.519779   330 leader_election.cc:240] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Election decided. Result: candidate lost.
I0701 17:10:11.519839  1067 raft_consensus.cc:2876] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 5.598s
I0701 17:10:11.519858  1067 raft_consensus.cc:2722] T c59a4e833445447aa25de7a34a5c03c7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Leader Pre-election lost for term 1. Reason: None given. Originator: 
I0701 17:10:11.775876  1068 raft_consensus.cc:813] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: ReportFailDetected: Starting NORMAL_ELECTION...
I0701 17:10:11.775921  1068 raft_consensus.cc:496] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Triggering leader pre-election, mode=NORMAL_ELECTION
I0701 17:10:11.775952  1068 raft_consensus.cc:2876] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 3.104s
I0701 17:10:11.776031  1068 raft_consensus.cc:535] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Starting pre-election with config: opid_index: -1 peers { permanent_uuid: "012245bfde494e12b856245c1b45022c" member_type: VOTER last_known_private_addr { host: "10.2.43.61" port: 9100 } last_known_broadcast_addr { host: "server1.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "35f6ba9096884be988424f4beb3353ea" member_type: VOTER last_known_private_addr { host: "10.2.80.72" port: 9100 } last_known_broadcast_addr { host: "server5.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "90825bc7b66c4dfbafa1eed3c39c1a3d" member_type: VOTER last_known_private_addr { host: "10.2.115.172" port: 9100 } last_known_broadcast_addr { host: "server2.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } }
I0701 17:10:11.776136  1068 leader_election.cc:215] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 35f6ba9096884be988424f4beb3353ea
I0701 17:10:11.776207  1068 leader_election.cc:215] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 90825bc7b66c4dfbafa1eed3c39c1a3d
W0701 17:10:11.776840   244 leader_election.cc:281] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 90825bc7b66c4dfbafa1eed3c39c1a3d: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: a68c9e1a7c0649bdbf91c63d78172775
W0701 17:10:11.777004   161 leader_election.cc:281] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 35f6ba9096884be988424f4beb3353ea: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: a68c9e1a7c0649bdbf91c63d78172775
I0701 17:10:11.777068   161 leader_election.cc:240] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Election decided. Result: candidate lost.
I0701 17:10:11.777127  1067 raft_consensus.cc:2876] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 4.408s
I0701 17:10:11.777153  1067 raft_consensus.cc:2722] T a68c9e1a7c0649bdbf91c63d78172775 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Leader Pre-election lost for term 1. Reason: None given. Originator: 
I0701 17:10:12.338863  1069 raft_consensus.cc:813] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: ReportFailDetected: Starting NORMAL_ELECTION...
I0701 17:10:12.338944  1069 raft_consensus.cc:496] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Triggering leader pre-election, mode=NORMAL_ELECTION
I0701 17:10:12.339005  1069 raft_consensus.cc:2876] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 4.131s
I0701 17:10:12.339166  1069 raft_consensus.cc:535] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Starting pre-election with config: opid_index: -1 peers { permanent_uuid: "671457e0cf824f1bab6eefc5a8788d71" member_type: VOTER last_known_private_addr { host: "10.2.101.52" port: 9100 } last_known_broadcast_addr { host: "server3.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "66ef2b92cb5244af8c5398f7f3838fdb" member_type: VOTER last_known_private_addr { host: "10.2.55.104" port: 9100 } last_known_broadcast_addr { host: "server4.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } } peers { permanent_uuid: "012245bfde494e12b856245c1b45022c" member_type: VOTER last_known_private_addr { host: "10.2.43.61" port: 9100 } last_known_broadcast_addr { host: "server1.yugabyte" port: 9100 } cloud_info { placement_cloud: "cloud1" placement_region: "NY2" placement_zone: "4152" } }
I0701 17:10:12.339282  1069 leader_election.cc:215] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 671457e0cf824f1bab6eefc5a8788d71
I0701 17:10:12.339411  1069 leader_election.cc:215] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Requesting vote from peer 66ef2b92cb5244af8c5398f7f3838fdb
W0701 17:10:12.340097   163 leader_election.cc:281] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 66ef2b92cb5244af8c5398f7f3838fdb: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: 742e6334d25d4c589ae3cf3c3eb721d7
W0701 17:10:12.340203   167 leader_election.cc:281] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 671457e0cf824f1bab6eefc5a8788d71: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: 742e6334d25d4c589ae3cf3c3eb721d7
I0701 17:10:12.340242   167 leader_election.cc:240] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Election decided. Result: candidate lost.
I0701 17:10:12.340323  1069 raft_consensus.cc:2876] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Snoozing failure detection for 3.350s
I0701 17:10:12.340355  1069 raft_consensus.cc:2722] T 742e6334d25d4c589ae3cf3c3eb721d7 P 012245bfde494e12b856245c1b45022c [term 0 FOLLOWER]: Leader Pre-election lost for term 1. Reason: None given. Originator: 
```

If you look at the timestamps, they are happening back to back.

**Comment by qrpike** [comment]
@rkarthik007 @Arnav15 Any ideas on how to solve this?

**Comment by rkarthik007** [comment]
Hi @qrpike,

Sorry missed responding to this. Here is one of the snippets:

```
W0701 17:10:11.201786   158 leader_election.cc:281] T 6b69a7a766604b1497dc13efab6a0a21 P 012245bfde494e12b856245c1b45022c [CANDIDATE]: Term 1 pre-election: Tablet error from VoteRequest() call to peer 35f6ba9096884be988424f4beb3353ea: Not found (yb/tserver/ts_tablet_manager.cc:1157): Tablet not found: 6b69a7a766604b1497dc13efab6a0a21
```

Basically, we're not finding the tablets. So I am wondering if the data directory for a subset of pods/services still has stale data from a previous incarnation of the universe. For example, the masters have retained the old data, but the tservers are newly deployed and do not have the tablets that the masters are referring to. Do you think this scenario is possible?

**Comment by Arnav15** [comment]
Hey @qrpike! Were you able to get up and running using @rkarthik007's suggestion? If not, let us know and we'll help you debug this further.

**Comment by rkarthik007** [comment]
Hi @qrpike - just checking if you were able to get up and running. Please feel free to reach out, someone from our side will be happy to get on a video call with you if that helps. I am closing this issue for now, please feel free to open a new issue or reopen this one.
